packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -642,56 +713,199 @@ abstract class TableBase extends Resource implements ITable {
+      });
+    }
+
+    const sum = new cloudwatch.MathExpression({
+      expression: `${Object.keys(usingMetrics).join(' + ')}`,
+      usingMetrics,
+      color: props?.color,
+      label: props?.label ?? 'Sum over all Operations',
+      period: props?.period,
+    });
+
+    return sum;
   }
 
   /**
-   * Metric for the user errors this table
-   *
-   * @default sum over a minute
+   * Metric for the user errors. Note that this metric reports user errors across all
+   * the tables in the account and region the table resides in.
    */
   public metricUserErrors(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
-    return this.metric('UserErrors', { statistic: 'sum', ...props });
+
+    if (props?.dimensions) {
+      throw new Error('"dimensions is not supported for the "UserErrors" metric');
+    }
+
+    // overriding 'dimensions' here because this metric is an account metric.
+    // see 'UserErrors' in https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/metrics-dimensions.html
+    return this.metric('UserErrors', { statistic: 'sum', dimensions: {}, ...props });
   }
 
   /**
    * Metric for the conditional check failed requests this table
-   *
-   * @default sum over a minute
    */
   public metricConditionalCheckFailedRequests(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
     return this.metric('ConditionalCheckFailedRequests', { statistic: 'sum', ...props });
   }
 
   /**
-   * Metric for the successful request latency this table
-   *
-   * @default avg over a minute
+   * Metric for the successful request latency for specific operations of this table
+   */
+  public metricSuccessfulRequestLatency(props: OperationalMetricOptions): cloudwatch.IMetric {
+
+    if (props?.dimensions?.Operation && props.operations) {
+      // these two represent the same thing, opted to require just one of them instead of dealing with merges
+      throw new Error("Operation can only be specified using the 'operations' property, not the Operation dimension.");
+    }
+
+    const operations = [];
+
+    if (props.dimensions?.Operation) {
+      // user specified the operation dimension, use it.
+      // note this means that props.operations is empty because of the above validation.
+      operations.push(props.dimensions?.Operation);
+    }
+
+    if (props.operations) {
+      // user specified the operations property, use it.
+      // note this means that the Operation dimension is not passed because of the above validation.
+      operations.push(...props.operations);
+    }
+
+    const usingMetrics: Record<string, cloudwatch.IMetric> = {};
+
+    const denominator = [];
+    const numerator = [];
+
+    if (operations.length > 5 || operations.length === 0) {
+      // you must pass at least 1 operation since this is an operation dimensionality metric.
+      // you cannot pass more than 5 because this would make this metric unusable for alarms who have a 10 individual metric limit for math expressions.
```

Comment:
!!!!!!!!!!!!!!!!!!! ARGLBARGL

Reply:
I know this is semi-out of scope but do you think you could see your way to adding a validation to `cloudwatch.Alarm` to that effect as well? Fairly sure this is going to come up more often and I'd rather not have everyone need to figure this out again and again.

Reply:
Added to alarm. I also removed this validation because creating graphs doesn't have this limitation. So the `Alarm` id definitely the correct place for it.